back/src/database/UserSQL.js

```diff
@@ -57,4 +57,40 @@ async function getUserByEmail(email) {
     });
 }
 
-module.exports = {saveUser:saveUser, getUserByEmail:getUserByEmail};
\ No newline at end of file
+async function edit(currentEmail, newEmail, newPassword, newUserName) {
+    var setQuery = "" 
+    if (newEmail != "") {
+        setQuery = setQuery + ` email='${newEmail}',`
+    }
+    if (newPassword != "") {
+        setQuery = setQuery + ` password='${newPassword}',`
+    }
+    if (newUserName!= "") {
+        setQuery = setQuery + ` username='${newUserName}',`
+    }
+    setQuery = setQuery.slice(0, -1)
+
+    const query = {
+        text: `UPDATE users.users SET ${setQuery} WHERE email='${currentEmail}'`
+    };
```

Comment:
Line 60 to 75 should be reviewed. We need the query to only update if the fields received are not null, undefined or "". The current version: newUserName!= "" is still updating the database. It should not.